lib3h/src/dht/dht_trait.rs

```diff
@@ -0,0 +1,19 @@
+use crate::dht::dht_event::{DhtEvent, PeerHoldRequestData};
+use lib3h_protocol::{Address, DidWork, Lib3hResult};
+
+/// Allow storage and retrieval of peer info and data
+///  + events & fns for gossip communication?
+pub trait Dht {
+    /// Peer info
+    fn get_peer(&self, peer_address: String) -> Option<PeerHoldRequestData>;
+    fn fetch_peer(&self, peer_address: String) -> Option<PeerHoldRequestData>;
+    fn drop_peer(&self, peer_address: String) -> Lib3hResult<()>;
+    /// Data
+    fn get_data(&self, data_address: Address) -> Lib3hResult<Vec<u8>>;
+    fn fetch_data(&self, data_address: Address) -> Lib3hResult<Vec<u8>>;
+    /// Processing
+    fn post(&mut self, evt: DhtEvent) -> Lib3hResult<()>;
```

Comment:
I think it makes sense to me that `post` takes ownership of the DhtEvent.

Above, though, I think the `peer_address` params should be `&str` and the `data_address` params should be `&AddressRef`

Reply:
oops forgot about that.